actions/docs-verifier/src/ActionRunner/Program.cs

```diff
@@ -23,20 +23,46 @@
 
 foreach (PullRequestFile file in files)
 {
-    if (file.IsRenamed() && file.PreviousFileName.EndsWith(".md", StringComparison.OrdinalIgnoreCase))
+    // Changing the extension from .yml to .md or the opposite doesn't require a redirection.
+    // In both cases, the URL in live docs site is the same.
+    if (file.IsRenamed() && !IsExtensionChangeOnly(file.PreviousFileName, file.FileName))
     {
         if (!await RedirectionsVerifier.WriteResultsAsync(Console.Out, file.PreviousFileName))
         {
             returnCode++;
         }
     }
-    else if (file.IsRemoved() && file.FileName.EndsWith(".md", StringComparison.OrdinalIgnoreCase))
+    else if (file.IsRemoved() && !files.Any(f => f.IsAdded() && IsExtensionChangeOnly(file.FileName, f.FileName)))
     {
         if (!await RedirectionsVerifier.WriteResultsAsync(Console.Out, file.FileName))
         {
             returnCode++;
         }
     }
+
+    static bool IsExtensionChangeOnly(string file1, string file2)
+    {
+        return TryStripYmlOrMarkdownExtension(file1, out string strippedFile1) &&
+            TryStripYmlOrMarkdownExtension(file2, out string strippedFile2) &&
+            strippedFile1.Equals(strippedFile2, StringComparison.OrdinalIgnoreCase);
+    }
```

Comment:
```suggestion
    static bool IsExtensionChangeOnly(string file1, string file2) =>
        TryStripYmlOrMarkdownExtension(file1, out string strippedFile1) &&
        TryStripYmlOrMarkdownExtension(file2, out string strippedFile2) &&
        strippedFile1.Equals(strippedFile2, StringComparison.OrdinalIgnoreCase);
```

Reply:
@IEvangelist `yaml` doesn't seem supported by docs build. See:

![image](https://user-images.githubusercontent.com/31348972/123991869-602bbe00-d9cb-11eb-8f58-6c137c4416aa.png)

vs

![image](https://user-images.githubusercontent.com/31348972/123991919-6c178000-d9cb-11eb-8833-83e16579f56c.png)

Build shows a preview link for `yml`, but not `yaml`. I'll adjust this and apply the suggestion.